### PR TITLE
Add country flags and mapping

### PR DESCRIPTION
**Pull request by kseha1080**

### CODE REVIEW

File: src/country.ts [hunk 1]
```diff
@@ -1,6 +1,497 @@
...
+import vnFlag from './static/flags/vn.svg';
+import vgFlag from './static/flags/vg.svg';
+import viFlag from './static/flags/vi.svg';
+import wfFlag from './static/flags/wf.svg';
+import ehFlag from './static/flags/eh.svg';
+import yeFlag from './static/flags/ye.svg';
+import zmFlag from './static/flags/zm.svg';
+import zwFlag from './static/flags/zw.svg';
+
+type CountryCode =
```
Comment (kseha1080) [inline src/country.ts:246]: @vincentaudoire Country code type created here

File: src/country.ts [hunk 2]
```diff
@@ -249,3 +740,250 @@ export const allCountries: Country[] = [
   { name: 'Zambia', code: 'ZM' },
   { name: 'Zimbabwe', code: 'ZW' },
 ];
+
+export const mapCountryFlag: { [K in CountryCode]: string } = {
```
Comment (kseha1080) [inline src/country.ts:744]: Mapping for flags by country code

File: src/country.ts [hunk 3]
```diff
@@ -1,6 +1,497 @@
...
+  | 'WF'
+  | 'EH'
+  | 'YE'
+  | 'ZM'
+  | 'ZW';
+
 export interface Country {
   name: string;
-  code: string;
+  code: CountryCode;
```
Comment (kseha1080) [inline src/country.ts:494]: Country interface is updated

**Review by vincentaudoire — CHANGES_REQUESTED** [review]

File: custom.d.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+declare module '*.svg' {
+  const content: any;
+  export default content;
+}
+
+declare module '\*.png' {
+  const content: string;
+  export default content;
+}
```
[the remark this answers is not in the document]
Comment (kseha1080) [reply custom.d.ts:9]: Seems like declaring these files here is needed for importing and using SVGs

File: src/country.ts (hunk 2 above)
Comment (kseha1080) [reply src/country.ts:744]: How would a function return images without having to write the conditions for each country code? Or would it just use this mapping?

Comment (vincentaudoire) [reply src/country.ts:744]: the function can use the mapping.

**Review by vincentaudoire — APPROVED** [review]

**Pull request merged**